app/Http/Controllers/BlackoutController.php

```diff
@@ -2,117 +2,165 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\AcademicDate;
+use Carbon\Carbon;
+use DateInterval;
+use DatePeriod;
+use DateTime;
 use Illuminate\Http\Request;
 use App\Models\Room;
 use App\Models\Blackout;
+use Illuminate\Validation\ValidationException;
 
 class BlackoutController extends Controller
 {
-    /**
-     * Display a listing of the resource.
-     *
-     * @return \Illuminate\Http\Response
-     
-
-    *public function index()
-    *{
-        
-    *}
-    */
-    /**
-     * Show the form for creating a new resource.
-     *
-     * @return \Illuminate\Http\Response
-     */
-
-    /**
-     * Store a newly created resource in storage.
-     *
-     * @param  \Illuminate\Http\Request  $request
-     * @return \Illuminate\Http\Response
-     */
-    public function store(Request $request)
-    {
-
-        $request->validateWithBag('createBlackout', [
-            'room_id' => ['required', 'integer', 'exists:'.Room::class.",id"],
-            'start' => ['required', 'date'],
-            'end' => ['required', 'date', "after:start"],
-            'name' => ['required','string']
-        ]);
-        
-        Blackout::create([
-            'start_time' => $request->start,
-            'end_time' => $request->end,
-            'name' => $request->name
-        ])->rooms()->attach($request->room_id);
+  /**
+   * Display a listing of the resource.
+   *
+   * @return \Illuminate\Http\Response
+   *public function index()
+   *{
+   *}
+   */
+  /**
+   * Show the form for creating a new resource.
+   *
+   * @return \Illuminate\Http\Response
+   */
+
+  /**
+   * Store a newly created resource in storage.
+   *
+   * @param \Illuminate\Http\Request $request
+   * @return \Illuminate\Http\Response
+   * @throws \Exception
+   */
+  public function store(Request $request)
+  {
+
+    $request->validateWithBag('createBlackout', [
+      'room_id' => ['required', 'integer', 'exists:' . Room::class . ",id"],
+      'start' => ['required', 'date'],
+      'end' => ['required', 'date', "after:start"],
+      'name' => ['required', 'string']
+    ]);
+
+    $academicDate = AcademicDate::query()
+      ->where('start_date', '<', Carbon::now())
+      ->where('end_date', '>', Carbon::now())
+      ->first();
+
+    switch ($request->get('recurring')) {
+      case'daily':
+        $startDatetime = Carbon::parse($request->start);
+        $endDatetime = Carbon::parse($request->end);
+
+        if (!$startDatetime->isSameDay($endDatetime)) {
+          throw ValidationException::withMessages(['recurrence_error' => 'Start date and End date cannot be different for recurring daily blackout']);
+        }
 
-        return back();
-    }
+        $beginLoop = new DateTime(Carbon::now()->format("Y-m-d"));
+        $endLoop = new DateTime($academicDate->end_date);
```

Comment:
Would be good practice to keep using Carbon and CarbonPeriod interfaces instead of the underlying